src/types.rs

```diff
@@ -363,123 +364,139 @@ impl From<&RoomSettings> for matrix_sdk_crypto::store::RoomSettings {
     }
 }
 
-/// Represent the type of {@link ProcessedToDeviceEvent}.
+/// The type of a {@link ProcessedToDeviceEvent}.
 #[wasm_bindgen]
 #[derive(Debug, Clone)]
 pub enum ProcessedToDeviceEventType {
-    /// A successfully-decrypted encrypted event.
+    /// A successfully-decrypted encrypted to-device message.
     Decrypted,
 
-    /// An encrypted event which could not be decrypted.
+    /// An encrypted to-device message which could not be decrypted.
     UnableToDecrypt,
 
-    /// An unencrypted event (sent in clear).
+    /// An unencrypted to-device message (sent in clear).
     PlainText,
 
-    /// An invalid to device event that was ignored because it is missing some
+    /// An invalid to-device message that was ignored because it is missing some
     /// required information to be processed (like no event `type` for
     /// example)
     Invalid,
 }
 
-/// Represent a ToDevice event after it has been decrypted by {@link
+/// Represent a to-device event after it has been decrypted by {@link
 /// OlmMachine#receiveSyncChanges}.
 #[wasm_bindgen(getter_with_clone)]
 #[derive(Debug, Clone)]
 pub struct DecryptedToDeviceEvent {
-    /// The type of processed event
-    #[wasm_bindgen(getter_with_clone, js_name = "type")]
-    pub processed_type: ProcessedToDeviceEventType,
-
-    /// A JSON-encoded string containing the processed event.
-    /// The decrypted event as if it was sent in clear (For room keys or secrets
-    /// some part of the content might have been zeroize'd).
-    #[wasm_bindgen(readonly, js_name = "wireEvent")]
-    pub wire_event: JsString,
+    /// A JSON-encoded object containing the processed event.
+    /// The decrypted message as if it was sent in clear (For room keys or
+    /// secrets some part of the content might have been zeroize'd).
+    /// Represents a Matrix to-device message with `type`, `sender` and
+    /// `content` fields.
+    #[wasm_bindgen(readonly, js_name = "decryptedRawEvent")]
+    pub decrypted_raw_event: JsString,
 
     #[wasm_bindgen(readonly, js_name = "encryptionInfo")]
-    /// The encryption information (olm variant) for the event.
-    pub encryption_info: crate::responses::EncryptionInfo,
+    /// The olm encryption information for the event.
+    pub encryption_info: OlmEncryptionInfo,
+}
+
+#[wasm_bindgen]
+impl DecryptedToDeviceEvent {
+    /// The type of processed to-device event. Always {@link
+    /// ProcessedToDeviceEventType.Decrypted} for this type.
+    #[wasm_bindgen(getter, js_name = "type")]
+    pub fn processed_type(&self) -> ProcessedToDeviceEventType {
+        ProcessedToDeviceEventType::Decrypted
+    }
+
+    /// Create a new `DecryptedToDeviceEvent`.
+    #[wasm_bindgen(constructor)]
+    pub fn new(decrypted_raw_event: JsString, encryption_info: OlmEncryptionInfo) -> Self {
+        Self { decrypted_raw_event, encryption_info }
+    }
```

Comment:
it's a bit odd to put the constructor *after* methods. Could you you put the constructor before `processed_type`, please.